dot/parachain/availability-distribution/availability_distribution.go

```diff
@@ -113,12 +121,40 @@ func (ad *AvailabilityDistribution) processAvailabilityDistributionMessageFetchP
 	return nil // TODO: implement
 }
 
-//nolint:unused
 func (ad *AvailabilityDistribution) handleChunkFetchingRequest(
-	who peer.ID,
+	_ peer.ID,
 	payload []byte,
 ) (network.ResponseMessage, error) {
-	return nil, nil // TODO: implement
+	request := &messages.ChunkFetchingRequest{}
+
+	err := request.Decode(payload)
+	if err != nil {
+		return nil, fmt.Errorf("decoding chunk fetching request: %w", err)
+	}
+
+	query := availabilitystore.QueryChunk{
+		CandidateHash:  request.CandidateHash,
+		ValidatorIndex: request.Index,
+		Sender:         make(chan availabilitystore.ErasureChunk),
+	}
+
+	ad.subSystemToOverseer <- query
+	response := &messages.ChunkFetchingResponse{}
+
+	// Ideally availability store would close the channel instead of sending an empty ErasureChunk.
+	// This would allow using `chunk, ok := <-query.Sender` and read `ok == false` as "chunk not found".
+	chunk := <-query.Sender
+	if chunk.Chunk == nil {
```

Comment:
We should consider both cases. Let's also check if the channel is open or closed.
```suggestion
	chunk, ok := <-query.Sender
	if !ok || chunk.Chunk == nil {
```